src/app/shared/apply-mask.directive.ts

```diff
@@ -0,0 +1,65 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+
+@Directive({
+  selector: '[applyMask]',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: ApplyMaskDirective,
+    multi: true,
+  }]
+})
+export class ApplyMaskDirective implements ControlValueAccessor {
+  onTouched: any;
+  onChange: any;
+
+  @Input('applyMask') applyMask: string;
+
+  writeValue(value: any): void { }
+
+  registerOnChange(fn: any): void {
```

Comment:
Man, use name more declarative for the vars, **fn**?? You could use **callback** for example
```suggestion
  registerOnChange(callback: any): void {
```